<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Long Branches Attract problem, and forcing a fix on it?</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Long Branches Attract problem, and forcing a fix on it?</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/375_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">BrianFoley</div>
          <div class="post_content">
<p>With HIV-1 phylogenies we quite often have data sets where we know important details about the true evolutionary history.  For one example, with a local transmission chain such as husband to wife to infant chain we can often know for certain who was infected first and the dates of transmission events etc.  In other cases the rough epidemiology is known so that we can tell that an epidemic spread out from a point source introduction.</p>
<p>Very often, or perhaps always given certain relative levels of diversity involved, the phylogenetic trees produced from a data set show a misrooting of one or more subclades which essentially turns that subclade “inside out” by putting the more diverse sequences rooted to the outgroup.  The explanation seems to be the “long branches attract problem”.  In the father-&gt;mother-&gt;infant case for example, the sequences from the infant often appear on a branch in between the father and mother, when we know for certain that this is “out of order”.</p>
<p>What I want is a tool that allows me to “fix” a known misrooting of a clade, and then calculate the likelihood value (or other such measurement) of the “correct” tree vs the misrooted tree.  I can provide sample data sets and tree results to anyone who is interested in this.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">funwithwords26</div>
          <div class="post_content">
<p>FastTree supports topological constraints, where you can force the final tree to have a certain split. It would probably work for this situation.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/381_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">josephwb</div>
          <div class="post_content">
<p>Right: I believe <em>all</em> programs offer such topological constraints. This seems to be what you want.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/293_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">prabhav_kalaghatgi</div>
          <div class="post_content">
<p>Hi Brian,</p>
<p>This is with reference to the following statement.</p>
<p>“In the father-&gt;mother-&gt;infant case for example, the sequences from the infant often appear on a branch in between the father and mother, when we know for certain that this is “out of order”.”</p>
<p>As you explained, this may be a result of long-branch-attraction. The following paper provides an alternate explanation. Please take a look at the cases that are presented in Fig. 1.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4137707/">
<header class="source">
<img class="site-icon" height="16" src="https://www.ncbi.nlm.nih.gov/favicon.ico" width="16"/>
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4137707/" rel="noopener nofollow ugc" target="_blank">PubMed Central (PMC)</a>
</header>
<article class="onebox-body">
<img class="thumbnail onebox-avatar" height="144" src="https://www.phylobabble.org/uploads/default/original/1X/3eeaa115acaa67b58b2325d28e2c2fca3542301a.png" width="144"/>
<h3><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4137707/" rel="noopener nofollow ugc" target="_blank">Timing and Order of Transmission Events Is Not Directly Reflected in a...</a></h3>
<p>Pathogen phylogenies are often used to infer spread among hosts. There is, however, not an exact match between the pathogen phylogeny and the host transmission history. Here, we examine in detail the limitations of this relationship. First, all...</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>Best,
Prabhav</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/375_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">BrianFoley</div>
          <div class="post_content">
<p>Here is an example tree, and a couple of figures drawn from it.  It is usually not the whole tree that is misrooted or “inside out”, but just one clade within the tree.  I am attaching here a tree that appears to have an example of the issue.  The HIV-1 M group subtype B epidemic began in Haiti and then there were hundreds of exports of the virus from Haiti to other parts of the world.  The sub-epidemic of subtype B introduction into South Korea happened in 1989 or 1990 and most likely did not come straight out of Haiti.  Subtype B spread into Trinidad and Tobago much earlier than 1990, but there was not much sampling and sequencing of viruses there until more recently, so there is a bit of a bias there for isolates in the post 2005 time period in comparison to USA/Europe where we have a lot of sequences from viruses sampled in the 1984 - 2000 time period.</p>
<p><div class="lightbox-wrapper"><a class="lightbox" data-download-href="https://www.phylobabble.org/uploads/default/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc" href="https://www.phylobabble.org/uploads/default/original/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc.jpg" title="MgroupTreeBRoots.jpg"><img data-small-upload="https://www.phylobabble.org/uploads/default/optimized/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc_2_10x10.png" height="500" src="https://www.phylobabble.org/uploads/default/optimized/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc_2_646x500.jpg" srcset="https://www.phylobabble.org/uploads/default/optimized/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc_2_646x500.jpg, https://www.phylobabble.org/uploads/default/optimized/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc_2_969x750.jpg 1.5x, https://www.phylobabble.org/uploads/default/original/1X/369dfdf3f73f869d80bacc8e1e07a71bf0e305bc.jpg 2x" width="646"/><div class="meta">
<svg aria-hidden="true" class="fa d-icon d-icon-far-image svg-icon"><use xlink:href="#far-image"></use></svg><span class="filename">MgroupTreeBRoots.jpg</span><span class="informations">1052×813 267 KB</span><svg aria-hidden="true" class="fa d-icon d-icon-discourse-expand svg-icon"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div></p>
<p>In this tree, we might get the impression that the viruses from South Korea and Trinidad/Tobago originate near the “root” of the subtype B clade which I have labeled as Root 1 in red.  But it is far more likely, given all that we know of the epidemiology of HIV-1 subtype B, that the true root of the B clade is close the the green node I have labeled as “root 2”.</p>
<p>When the tree is drawn as a radial tree like this, it gives us a bit of a different impression that when it is drawn as a cladogram.  And in either tree view, if we measure distances from tips to “root 1” we get different values than if we measure distances from tips to “root 2”.</p>
<p><div class="lightbox-wrapper"><a class="lightbox" data-download-href="https://www.phylobabble.org/uploads/default/df4c152f69d56daff765bfd2255f57a1cd3a2b59" href="https://www.phylobabble.org/uploads/default/original/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59.jpg" title="MgroupTreeCladogramLabeled.jpg"><img data-small-upload="https://www.phylobabble.org/uploads/default/optimized/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59_2_10x10.png" height="500" src="https://www.phylobabble.org/uploads/default/optimized/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59_2_312x500.jpg" srcset="https://www.phylobabble.org/uploads/default/optimized/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59_2_312x500.jpg, https://www.phylobabble.org/uploads/default/optimized/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59_2_468x750.jpg 1.5x, https://www.phylobabble.org/uploads/default/optimized/1X/df4c152f69d56daff765bfd2255f57a1cd3a2b59_2_624x1000.jpg 2x" width="312"/><div class="meta">
<svg aria-hidden="true" class="fa d-icon d-icon-far-image svg-icon"><use xlink:href="#far-image"></use></svg><span class="filename">MgroupTreeCladogramLabeled.jpg</span><span class="informations">877×1404 218 KB</span><svg aria-hidden="true" class="fa d-icon d-icon-discourse-expand svg-icon"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div></p>
<p>The chat software here does not allow me to upload the treefile or alignment, but I can send it to anyone who wants it.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/375_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">BrianFoley</div>
          <div class="post_content">
<p>Another example.  <div class="lightbox-wrapper"><a class="lightbox" data-download-href="https://www.phylobabble.org/uploads/default/b4050cbd1feff0dcdbcfde806a17b16d9683a913" href="https://www.phylobabble.org/uploads/default/original/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913.jpg" title="Fungus_ITS_treeJPG.jpg"><img data-small-upload="https://www.phylobabble.org/uploads/default/optimized/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913_2_10x10.png" height="500" src="https://www.phylobabble.org/uploads/default/optimized/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913_2_278x500.jpg" srcset="https://www.phylobabble.org/uploads/default/optimized/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913_2_278x500.jpg, https://www.phylobabble.org/uploads/default/optimized/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913_2_417x750.jpg 1.5x, https://www.phylobabble.org/uploads/default/optimized/1X/b4050cbd1feff0dcdbcfde806a17b16d9683a913_2_556x1000.jpg 2x" width="278"/><div class="meta">
<svg aria-hidden="true" class="fa d-icon d-icon-far-image svg-icon"><use xlink:href="#far-image"></use></svg><span class="filename">Fungus_ITS_treeJPG.jpg</span><span class="informations">785×1407 424 KB</span><svg aria-hidden="true" class="fa d-icon d-icon-discourse-expand svg-icon"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div></p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>