<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Postdoc position in statistical phylogenetics/data analysis</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Postdoc position in statistical phylogenetics/data analysis</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">stephaneguindon</div>
          <div class="post_content">
<p>. <strong>Project  title</strong>:  Bayesian methods  and graphical  approaches to improve our understanding of molecular evolution using mutation maps.
. <strong>Principal investigator</strong>: <em>Stephane Guindon</em> (CNRS)
. <strong>Associate  investigators</strong>: <em>Nicolas  Galtier</em> (CNRS),  <em>Fabio Pardi</em> (CNRS), <em>Anne-Muriel Chifolleau</em> (UM),  <em>Francois Chevenet</em> (IRD) &amp; <em>Anne-Laure Banuls</em> (IRD).
. <strong>Host institution</strong>: CNRS-University of Montpellier, France.
. <strong>Funding</strong>:  NUMEV (Digital and Hardware  Solutions, Environmental and Organic Life Modeling).
. <strong>Net salary</strong>: ~2100 euros/month.
. <strong>Starting date</strong>: no later than April 2017.</p>
<p>A  18  month postdoc  position  in  statistical phylogenetics  is available  at the  University  of Montpellier  (UM), France.  The successful  candidate will  join the  “Methods and  Algorithms in Bioinformatics” group and work  in close collaboration with other Montpellier-based research teams (see  above for more information about the personnel involved).</p>
<p>The  project  relies on  Bayesian  sampling  techniques that  map mutations on phylogenies [1].  These approaches are very relevant from  a biological  [2,3]  and a  computational [4]  perspective. However,  we  believe that  their  full  potential has  not  been reached  yet.  In  particular,  applying  novel data  exploration techniques to visualize mutation maps  should help us improve our understanding of  the fine  scale mechanisms  governing molecular evolution.</p>
<p>The  postdoc candidate  will  thus  develop original  information visualization methods that will provide relevant summaries of the mutational  process. The  proposed  techniques  will also  detect episodes of evolution  that are not well predicted  by our Markov models  of evolution.   Applications of  these new  visualization techniques  are  manifold.  Adaptive evolution  and  biased  gene conversion  are associated  with  peculiar substitution  patterns that mutation  maps could  potentially help  identify. Similarly, errors   in   sequence   alignment  and/or   improper   orthology relationships  are expected  to generate  atypical maps.  Lastly, preliminary  analyses   of  Mycobacterium   tuberculosis  genomes demonstrated how  mutation maps  can help  recover the  series of mutations  leading  to   antibiotic  resistance.  The  successful candidate will further  develop one or several  of these research leads.</p>
<p>The ideal candidate  for the proposed project will have  a PhD in statistical phylogenetics  or population genetics,  although pure statisticians and  physicists with  strong interest  in molecular evolution should  also apply. Good  skills in data  analysis with modern tools  and programming  languages (R and/or  Python and/or Java or C/C++) are essential.</p>
<p>Please   send    CVs   and   inquiries   to    Stephane   Guindon (<a href="mailto:guindon@lirmm.fr">guindon@lirmm.fr</a>).</p>
<p>Bibliography:</p>
<ul>
<li>
<p>[1]  Nielsen, R.  Mapping  mutations  on phylogenies.  Systematic biology. 51. 729-739. 2002.</p>
</li>
<li>
<p>[2] Dutheil  J, Pupko T,  Jean-Marie A, Galtier N.  A model-based approach    for   detecting    co-evolving    positions   in    a molecule. Molecular biology and evolution. 22. 1919-1928. 2005.</p>
</li>
<li>
<p>[3]  Dutheil  J,  Galtier  N.  Detecting  groups  of  co-evolving positions in a molecule: a clustering approach.  BMC evolutionary biology. 7. 2007.</p>
</li>
<li>
<p>[4]  Rodrigue N.,  Philippe H.,  Lartillot N.  Uniformization for sampling  realizations  of   Markov  processes:  applications  to Bayesian      implementations      of     codon      substitution models. Bioinformatics. 24. 2007.</p>
</li>
</ul>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>