<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Prunning off long branches</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Prunning off long branches</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">kmkozak87</div>
          <div class="post_content">
<p>Dear All,</p>
<p>I am building a pipeline to automatically generate gene trees for about 10,000 CDS alignments (all genes from an exome). The genes were sequenced for 150 individuals in multiple species. Some individuals are worse than others and occasionally have little data in some alignments, and end up on obviously artificially inflated branches. Is anyone aware of a tool to prune those automatically?
(I will also use tools to get rid of poor sequence first, but that’s a different topic.)</p>
<p>Many thanks,
Krzysztof Kozak</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/172_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">lpryszcz</div>
          <div class="post_content">
<p>Hi Krzysztof,</p>
<p>Have a look at <a href="http://etetoolkit.org/" rel="nofollow noopener">ETE toolkit</a> for tree handling (Python based). You will have to write simple script that recognises and prunes long branches. <a href="https://www.biostars.org/p/74459/#74485" rel="nofollow noopener">Here</a> you have a pruning example.</p>
<p>You may want to look at <a href="http://nar.oxfordjournals.org/content/39/suppl_1/D556.full" rel="nofollow noopener">phylomeDB phylogenetic tree reconstruction pipeline</a> for some ideas how to improve alignments by using multiple aligners and trim poorly aligned regions automatically. Finally, the way to improve the phylogenetic reconstruction itself are also given.</p>
<p>Good luck!</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">yangya</div>
          <div class="post_content">
<p>I wrote a simple python script to pruning branches longer than a certain threshold. See the script “final_cut_long_branched.py” in <a href="https://bitbucket.org/yangya/phylogenomic_dataset_construction" rel="nofollow noopener">https://bitbucket.org/yangya/phylogenomic_dataset_construction</a>. The tree libraries phylo3.py and newick3.py are also included in the same repository, which were written by Stephen Smith.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>