<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>p4 Python phylogenetics toolkit</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">p4 Python phylogenetics toolkit</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>News to me: <a href="http://p4.nhm.ac.uk/index.html">p4 phylogenetics toolkit</a>, which does inference and manipulation.</p>
<p>P4 is a Python package for maximum likelihood and Bayesian analysis of molecular sequences. Its specialty is that it can use heterogeneous models, where the characteristics of the model can differ over the data or over the tree.</p>
<ul>
<li>P4 can be used as a phylogenetic toolkit, the elements of which you can string together in different ways depending on the job at hand. It is useful for programmatic manipulation of phylogenetic data and trees. If you want to do something interesting with your trees or data, p4 might have at least some of what you want to do already in place.</li>
<li>P4 will read data in a few of the common phylogenetic formats (eg Nexus, Phylip, clustalw, fasta, pir/nbrf), but does not read other formats in bioinformatics (eg EMBL, genbank). P4 will read in trees in Nexus or Phylip format.</li>
<li>P4 will do some elementary data manipulation, eg extracting a Nexus-defined charset from an alignment, or converting data from one format to another. P4 will also do tree manipulation, and tree drawing.
It has a big tree viewer, to be able to view big trees (eg up to 5000 taxa) on the screen.</li>
<li>P4 is meant to be easily extensible, so if you want to do something that it cannot do, it is often easy to add that functionality.</li>
</ul>
<p>As far as I can tell, there isn’t an overlap with <span class="mention">@jeetsukumaran</span>’s DendroPy or Paul Lewis’ Phycas.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>