<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Mike Steel&rsquo;s 2011 predictions&ndash; how is he doing so far?</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Mike Steel&rsquo;s 2011 predictions&ndash; how is he doing so far?</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Every five years, Mike Steel puts predictions up on his <a href="http://www.math.canterbury.ac.nz/~m.steel/">webpage</a> of five directions in phylogenetics that will grow in the next five years.</p>
<p>His predictions for the period 2011-2016:</p>
<ol>
<li>Development of network­‐based methods to display ‘evolution as it happened’ including reticulation (LGT, endosymbiosis, hybrid species etc) up to the limits of what can be discerned from extant data.</li>
<li>Phylogenetic approaches for handling patchy taxon coverage and analyzing large numbers of short reads from next generation sequencing.</li>
<li>Phylogenetic approaches to early life using non­‐stationary models and protein structural constraints</li>
<li>Statistical approaches for analyzing non­‐aligned sequence data.</li>
<li>More realistic models of speciation and extinction that better describe the shape of ‘real’ phylogenies.</li>
</ol>
<p>I can say that people are certainly <em>doing</em> <span class="hashtag">#2</span>, but it seems to me that most folks are just running classical approaches to datasets that are a high percent gap.</p>
<p>A lot of these touch on topics that I don’t know anything about, such as early life.</p>
<p>Thoughts, anyone?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/138_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">rob_lanfear</div>
          <div class="post_content">
<p>I was just at the NZ phylogenetics meeting with Mike Steel and a bunch of others. My general feeling from that meeting (abstracts here: <a href="http://www.math.canterbury.ac.nz/bio/events/waiheke2014/files/programme.pdf" rel="nofollow noopener">http://www.math.canterbury.ac.nz/bio/events/waiheke2014/files/programme.pdf</a>) on these points is:</p>
<ol>
<li>Some interesting work here, and in particular an awesome talk on figuring out the limits of what can be discerned from extant data.</li>
<li>Not much new here. In fact, it strikes me that patchy coverage is still an open question (even to the point of there not having a great picture from simulations). Separate to the NZ meeting, I know that some folks in Allen Rodrigo’s group at Duke are doing awesome work on using short reads directly (rather than mapping them first and calling genotypes), and integrating that into BEAST.</li>
<li>Non stationary models still seems like a sticking point. Though there was one talk at the NZ meeting which proposed an interesting model for this in which the amino acid matrices grow over time as new amino acids are added to the code. (With empirical analyses too…)</li>
<li>Nothing from the meeting. But Ben Redelings (Duke / NESCent) is still working on BALi-phy which does this. Anyone know of any other work?</li>
<li>Lots of this. Usually involving Tanja Stadler from ETH Zurich.</li>
</ol>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Funny, when I first read <span class="hashtag">#4</span> I thought he was talking about</p>
<aside class="onebox wikipedia">
<header class="source">
<a href="https://en.wikipedia.org/wiki/Alignment-free_sequence_analysis" rel="nofollow noopener" target="_blank">en.wikipedia.org</a>
</header>
<article class="onebox-body">
<h3><a href="https://en.wikipedia.org/wiki/Alignment-free_sequence_analysis" rel="nofollow noopener" target="_blank">Alignment-free sequence analysis</a></h3>
<p>In bioinformatics, alignment-free sequence analysis approaches to molecular sequence and structure data provide alternatives over alignment-based approaches. The emergence and need for the analysis of different types of data generated through biological research has given rise to the field of bioinformatics. Molecular sequence and structure data of DNA, RNA, and proteins, gene expression profiles or microarray data, metabolic pathway data are some of the major types of data being analysed in bioi...</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>and I’m actually not sure now. Let’s see if we can get him to join the forum.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/15_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mathmomike</div>
          <div class="post_content">
<p>Actually a very interesting (theoretical) paper on this has just appeared in Ann. Appl. Prob. by Daskalakis and Roch ("Alignment-free phylogenetic reconstruction: sample complexity via a branch process analysis)
see
<a class="onebox" href="https://projecteuclid.org/euclid.aoap/1360682027" rel="nofollow noopener" target="_blank">https://projecteuclid.org/euclid.aoap/1360682027</a></p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/376_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mlandis</div>
          <div class="post_content">
<aside class="quote" data-post="2" data-topic="48">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/rob_lanfear/40/138_1.png" width="20"/> rob_lanfear:</div>
<blockquote>
<p>Nothing from the meeting. But Ben Redelings (Duke / NESCent) is still working on BALi-phy which does this. Anyone know of any other work?</p>
</blockquote>
</aside>
<p>Bouchard-Côté and Jordan’s <a href="http://www.pnas.org/content/110/4/1160.short" rel="nofollow noopener">Poisson Indel Process</a> is similar to the approach described in <a href="http://link.springer.com/article/10.1007/BF02193625" rel="nofollow noopener">TKF91</a>, where the process allows indel events to forgo the need of an alignment, while still remaining tractable for large numbers of taxa. I haven’t seen it in action, though.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Yes, for those interested in the PIP, Bouchard-Côté described in a phyloseminar:</p>
<div class="lazyYT" data-height="270" data-parameters="feature=oembed&amp;wmode=opaque" data-width="480" data-youtube-id="ycqAUcra0Ys" data-youtube-title="The Poisson Indel Process"></div>
<p> </p>
<p><span class="mention">@mathmomike</span> if this is the same as <a href="http://arxiv.org/abs/1109.5002">the version of their paper on arxiv</a> then this is a distance-based algorithm, not a likelihood-based one, right? So would that satisfy your 2011 criteria?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/15_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mathmomike</div>
          <div class="post_content">
<p>Yes it’s distance-based, but it is still ‘statistical’ - i.e. they prove consistency and convergence results. It might not be as efficient as ML, but at least it is tractable!</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/383_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">boussau</div>
          <div class="post_content">
<aside class="quote" data-post="2" data-topic="48">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/rob_lanfear/40/138_1.png" width="20"/> rob_lanfear:</div>
<blockquote>
<p>Non stationary models still seems like a sticking point. Though there was one talk at the NZ meeting which proposed an interesting model for this in which the amino acid matrices grow over time as new amino acids are added to the code. (With empirical analyses too…)</p>
</blockquote>
</aside>
<p>I could not find the abstract that corresponds to your description. Do you remember the name of the authors? I’m intrigued! Thanks!</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/138_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">rob_lanfear</div>
          <div class="post_content">
<p>Hmmm. I can’t find it either. It must have been a replacement talk. <span class="mention">@mathmomike</span> can you remember? It’s the NZ physicist at Cantebury (I think) who works on models of amino acid replacement that account for the addition of new amino acids…</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/15_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mathmomike</div>
          <div class="post_content">
<p>It was Assoc. Prof. Peter Wills, a physicist from Auckland University; his website is
<a class="onebox" href="https://www.physics.auckland.ac.nz/en/about/our-staff/associate-professor-peter-wills.html" rel="nofollow noopener" target="_blank">https://www.physics.auckland.ac.nz/en/about/our-staff/associate-professor-peter-wills.html</a></p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/62_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">martinkova</div>
          <div class="post_content">
<ol>
<li>We developed a signature-based method for searching for horizontal gene transfer in Eukaryotes, called SigHunt (<a href="http://bioinformatics.oxfordjournals.org/content/early/2014/02/06/bioinformatics.btt727" rel="nofollow noopener">Jaron et al. 2014. Bioinformatics</a>). I was fighting against self-promotion here for the few days, but now we’re looking for collaborators for an upcoming grant application. If you’re interested in a specific organism in this respect, let me know and we could discuss it.</li>
</ol>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">funwithwords26</div>
          <div class="post_content">
<p>I have some doubts about “phylogenetic approaches for … analyzing large numbers of short reads from next generation sequencing.” I think short read data is generally low quality data and anything that can be done differently should be. Even for environmental samples there are starting to be alternatives to big piles of unassembled short reads, such as better metagenomics assembly tools, longer read technologies, or Illumina’s Moleculo,</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/71_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">peterwills</div>
          <div class="post_content">
<p>You can read a paper on the background to the approach I am taking to the phylogeny of the coding enzymes (and hope to apply to other ancient structures) using increasing-size alphabets at:</p>
<p>Wills PR. Genetic information, physical interpreters and thermodynamics; the material-­informatic basis of biosemiosis. Biosemiotics DOI 10.1007/s12304-013-9196-2 (online, 5 October 2013)</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/383_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">boussau</div>
          <div class="post_content">
<p>Thanks very much for this reference!</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/71_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">peterwills</div>
          <div class="post_content">
<p>The pagination is now available: Biosemiotics 7, 141–165 (2014).</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">alexei_drummond</div>
          <div class="post_content">
<p>That link seems broken now. Here is one I found that works:
<a class="onebox" href="http://www.science.auckland.ac.nz/people/p-wills" rel="nofollow noopener" target="_blank">http://www.science.auckland.ac.nz/people/p-wills</a></p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p><span class="mention">@mathmomike</span> – it seems like these predictions are an every-5-year event. Are we going to get some new ones?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/15_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mathmomike</div>
          <div class="post_content">
<p>Hi Erick,</p>
<p>good point - yes, it’s always in April, so I have only one more week to go…  I’ll upload something before the end of the month!  Any suggestions meantime most welcome <img alt=":wink:" class="emoji" src="//www.phylobabble.org/images/emoji/twitter/wink.png?v=5" title=":wink:"/></p>
<p>best,
Mike</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Well, if it’s not already obvious, I think that in 5 years we’re going to have Bayesian samplers that are based on foundations other than vanilla MCMC. I also think that we’ll see new and better methods to extend existing inferences with more taxa.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/117_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">argriffing</div>
          <div class="post_content">
<p>“the best way to predict the future is to invent it”!</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>