<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>How to infer topology and clade support with very low diversity alignments?</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">How to infer topology and clade support with very low diversity alignments?</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>I was reading a paper recently that did a full-on BEAST analysis on what appeared to be an alignment with very few informative columns-- independent relaxed clocks, skyline plots, etc. It raised my eyebrows.</p>
<p>But what is best practice with that sort of alignment? My sense is that because there is little data, parsimony would be best. The only paper I could think of, though, was <a href="http://sysbio.oxfordjournals.org/content/49/4/817.full.pdf+html">Sanderson and Kim’s paper</a> on “parametric phylogenetics.”</p>
<p>What about clade support? Non-parametric bootstrapping seems a bit strange because of it equates the empirical distribution on columns with the distribution generated by evolutionary processes. I suppose parametric bootstrapping would be good, conditioning on the number of informative sites to be the same as in the original alignment?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/376_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mlandis</div>
          <div class="post_content">
<p>Thanks for the paper. Looks interesting.</p>
<p>For your informative site bootstrapping approach, say you noticed informative sites occurred only at some third position codon sites and nowhere else. In this case I’d be inclined to generate bootstrap samples from third position codon sites rather than from informative sites. This would introduce some sampling error into the number of informative sites per bootstrap replicate (my intuition is this would weaken clade support values).</p>
<p>On a side note, I think a Bayesian approach could work well with due scrutiny given to the prior and model specification (e.g. model testing, prior sensitivity analysis). If the posterior resembles the prior for a broad class of models and priors, I’d take this as a red flag that there may not be enough data to resolve the tree.</p>
<p>There must be some low-data threshold at which there is no power to identify the tree. Since sequencing is so cheap, in many cases one might address the problem (not your question) by generating more data!</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">david_bryant</div>
          <div class="post_content">
<p>At first I thought that you might be trying to fashion a purse out of a sow’s ear, but now I’m wondering.</p>
<p>Could it be that having a only a few informative sites (and I assume, a large number of constant sites) could say <em>more</em> about, say, topology that many more ‘faster rate’ sites. Because the rate is supposedly low, backward mutations are highly unlikely, so that any informative sites are likely to nail a split in the tree. It could be that the posterior distribution (perhaps on multifurcating trees?) is more spiked than in other circumstances.</p>
<p>A single, clear, fingerprint on the lead piping is more informative than a potential weapon covered with fingerprints…</p>
<p>Is there some way that this can be formalised?</p>
<p>-David.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<aside class="quote" data-post="2" data-topic="32">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/mlandis/40/118_1.png" width="20"/> mlandis:</div>
<blockquote>
<p>In this case I’d be inclined to generate bootstrap samples from third position codon sites rather than from informative sites.</p>
</blockquote>
</aside>
<p>I think we are on the same page, but I’m not sure, so just to make sure:</p>
<ul>
<li>By informative sites in this setting I mean sites that are non-constant.</li>
<li>The idea would be to generate a parametric bootstrap sample of the same alignment length as before, but conditioned on having the same number of informative (non-constant) sites as in the original alignment.</li>
<li>I was thinking of doing a nucleotide tree (parsimony or IID model) which wouldn’t respect position.</li>
</ul>
<p>WRT Bayesian methods, are there any phylogenetic methods that perform shrinkage towards topological multifurcations? That would seem useful in this regime.</p>
<aside class="quote" data-post="2" data-topic="32">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/mlandis/40/118_1.png" width="20"/> mlandis:</div>
<blockquote>
<p>There must be some low-data threshold at which there is no power to identify the tree. Since sequencing is so cheap, in many cases one might address the problem (not your question) by generating more data!</p>
</blockquote>
</aside>
<p>Yes, but I think having methods to assess whether there is enough data would be useful. Of course we always need more data, but even if sequencing was free, sample collection and prep will never be.</p>
<p><span class="mention">@david_bryant</span> seems like (ahem) a simulation study would be useful here.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/376_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">mlandis</div>
          <div class="post_content">
<aside class="quote" data-post="4" data-topic="32">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/ematsen/40/114_1.png" width="20"/> ematsen:</div>
<blockquote>
<p>WRT Bayesian methods, are there any phylogenetic methods that perform shrinkage towards topological multifurcations? That would seem useful in this regime.</p>
</blockquote>
</aside>
<p><a href="http://sysbio.oxfordjournals.org/content/54/2/241.long" rel="nofollow noopener">Lewis, Holder, and Holsinger (2005)</a> describe one way to assign priors to polytomies vs. bifurcations, using rjMCMC to move through the space of trees with varying numbers of polytomies. It looks like it’s implemented in <a href="http://hydrodictyon.eeb.uconn.edu/projects/phycas/index.php/Download" rel="nofollow noopener">Phycas</a> if you’re curious to try it.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/141_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">blackrim</div>
          <div class="post_content">
<p>Great conversation. Coincidentally, Joseph Brown and I are doing just this right now (basically looking at methods for seeing if there is enough data to estimate a branch). Should have some code and write up soon. But completely agree. Especially when looking at large phylogenies (thousands of tips) this is a necessary operation.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/381_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">josephwb</div>
          <div class="post_content">
<p>Very cool how thinking about these kind of questions is aligning with old school parsimony papers that were dealing with extremely small data sets (and hence, the information content of, and compatibility amongst, expensive individual characters was of interest). With the sea of data currently available, seems like the assumption is <em>the signal will win out</em>. Where that signal comes from, and where it is absent, seems to have fallen from interest.</p>
<p>Of course, with these large empirical alignments we have the additional concern of data coverage, formalized by [Sanderson &amp; Steel’s][1] concept of “decisiveness”. Does the distribution of data coverage allow the recovery of specific bipartitions (without some sort of extrapolation from the model)?</p>
<p>So, seems to me this is two-pronged: 1) what bipartitions can <em>possibly</em> be reconstructed?, and 2) of those (conflicting) bipartitions, which of those are supported by the data (and by which characters)?
[1]: <a href="http://www.biomedcentral.com/1471-2148/10/155/" rel="nofollow noopener">http://www.biomedcentral.com/1471-2148/10/155/</a></p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p><span class="mention">@josephwb</span> yes. Note that there are very important current applications with the best of today’s technology and 100% coverage. For example, take the recent case of <a href="http://stm.sciencemag.org/content/4/148/148ra116.abstract">tracking hospital infections using whole genome sequencing</a>. Sequence the whole genome of the bugs and you get…</p>
<p><img alt="" height="" src="https://www.phylobabble.org/uploads/default/3/6627d9921d39d319.jpg" width=""/></p>
<p><span class="mention">@david_bryant</span> when are you going to tell us to use <a href="http://www.beast2.org/wiki/index.php/SNAPP">SNAPP</a>?</p>
<p>Are there any electronic copies of the <em>Parametric bootstrapping  in molecular phylogenetics: Applications and performance</em> paper out there? Was cited in <span class="mention">@arambaut</span>’s original Seq-Gen paper.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/121_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">trvrb</div>
          <div class="post_content">
<p>This may be completely naive, but I often find it useful to understand what should be straight-forward things by running simple visual experiments.  Here, I’m trying to understand the difference between a bootstrap interval and a proper posterior credible interval when the amount of data is small. (I’m sure this has been formally done by others…)</p>
<p>I’m using a simple Bernoulli model with a true underlying <em>p</em> of 0.5.  In this case, I make 100 observations from this distribution and observe 47 successes.  The point estimate of <em>p</em> is 0.47.  I show both a bootstrap resampling (in black) and a Bayesian posterior (in red) assuming a uniform prior on <em>p</em>.</p>
<p><img height="273" src="https://www.phylobabble.org/uploads/default/14/6c556683bc2dd3d7.png" width="417"/> </p>
<p>We see that the bootstrap distribution and the Bayesian posterior distribution are pretty much identical.  However, if I reduce <em>n</em> to 4 representing little data and observe 2 successes, I get the following:</p>
<p><img height="274" src="https://www.phylobabble.org/uploads/default/13/b645a921514e54eb.png" width="417"/> </p>
<p>It looks like the bootstrap distribution is wider than the Bayesian posterior distribution.</p>
<p>I swear I’ve read that Bayesian clade credibility intervals <em>overestimate</em> clade support (I’m assuming in relation to bootstrap clade confidence intervals).  Is this possibly the same basic effect as I’m showing in the simple Bernoulli example?  Although in this example, it looks like the bootstrap with little data is giving overly wide confidence intervals.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/138_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">rob_lanfear</div>
          <div class="post_content">
<p>I don’t necessarily think that this kind of analysis is a problem, as long as (as others have said) the authors make a serious attempt to figure out what’s coming from the data and what’s coming from the priors. Just running the analysis without the data (i.e. all sites changed to gaps) would let you sample from the real priors. Nicolas Lartillot has some interesting posts on his blog that are relevant (start here and work forwards: <a href="http://www.bayesiancook.blogspot.fr/2014/01/the-myth-of-over-parameterization.html" rel="nofollow noopener">http://www.bayesiancook.blogspot.fr/2014/01/the-myth-of-over-parameterization.html</a>), which describe why massive overparameterisation is often not an issue for Bayesian stats.</p>
<p>There are some related issues with clade support in Bayesian analyses that may be relevant here (e.g. the star-tree paradox), which Ziheng Yang (and others) have worked on a fair bit: <a href="http://mbe.oxfordjournals.org/content/24/8/1639.short" rel="nofollow noopener">http://mbe.oxfordjournals.org/content/24/8/1639.short</a>.</p>
<p>I guess best practice depends on the question. And at first glance I agree that very few informative columns is not a great start…</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Thanks for your thoughtful response.</p>
<p>With respect to <span class="mention">@nicolas_lartill</span>’s blog post, I don’t think that even he would say that “massive overparameterisation is often not an issue for Bayesian stats.” The point of that post was that <em>if we set things up right with regularization</em> problems with over-parameterization can be avoided. Here, as <span class="mention">@mlandis</span> points out, the relevant regularization is a way to contract bifurcations to multifurcations, and that requires a complex rjMCMC not implemented in BEAST.</p>
<p>With respect to the star tree paradox (high clade support even when there is no clade structure whatsoever), my impression was that it is a real problem that <a href="http://mbe.oxfordjournals.org/content/24/4/1075.full">does not go away</a> even in the case of lots of data, which is a better situation than the one I describe. This further motivates a way to return multifurcations as an end result.</p>
<p>I guess that to me, running an rjMCMC when you have an alignment like I describe above just seems like overkill.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/138_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">rob_lanfear</div>
          <div class="post_content">
<p>Yes, sorry - lazy hyperbole on my part to say ‘massive overperameterisation is OK’.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/16_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">nicolas_lartill</div>
          <div class="post_content">
<p>Hi All,</p>
<p>(and thanks for the invitation!)</p>
<p>I think that, in principle, both maximum likelihood and Bayesian analysis on small, low-diversity alignments (few non-constant columns) will not behave too badly, in the sense that you should not get strong support for wrong clades in this regime.</p>
<p>But then, most often, I guess you will just not get strong support for any clade. So, at the end of the analysis, you are most often supposed to conclude that the alignment was indeed too small to be of any good.</p>
<p>However, I can imagine that, in many cases, people will instead correct for small sample size and will start to interpret patterns that have insufficiently strong support: they will take clades that have pp&gt;0.7 seriously, for instance. Yet, those clades are probably highly unstable: they may just be statistical fluctuations.</p>
<p>So, perhaps, the question should not be: which method will be suited to analyzing very small alignments? the question should perhaps more fundamentally be: what are the limits there, in terms of minimum sample size and minimum statistical support, below which we can consider that we are just in a fundamentally unreliable regime, whichever method we use?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/138_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">rob_lanfear</div>
          <div class="post_content">
<p>I’d be interested to hear people’s opinions here, but for what it’s worth, I think we can quite easily figure out the inferential limits on any given dataset, as long as we can frame alternative hypotheses in terms of the topologies they generate.</p>
<p>Let’s imagine that we have two hypotheses that generate different sets of trees of size N and M respectively. In a Bayesian framework we could just analyse the posterior distribution of trees and see how frequently N and M appear in the prior (i.e. no data) versus the posterior (i.e. with data) and use that to estimate a Bayes factor. There could be issues if N or M is small, but these are not intractable, e.g. we could only sample from sets N and M in the prior (simple if N and M are easy to specify as constraints, a bit harder otherwise).</p>
<p>Similarly, in an ML framework we could e.g. estimate the ML tree from set N and set M, and then use e.g. SOWH tests (i.e. parametric bootstraps) to simulate data and ask whether the overall ML tree (i.e. best of N and M) is sufficiently good to reject the alternative hypothesis.</p>
<p>My suspicion is that the Bayesian approach (or some modification) would be best when information content is low, because it seems odd to consider only a single ML tree (as in the SOWH test) if there’s very little information on topology.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/16_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">nicolas_lartill</div>
          <div class="post_content">
<p>yes, I agree that the Bayesian method should make a more efficient use of the available information by averaging over “nuisance parameters” (here averaging over trees). On the other hand, it does that by integrating over the prior, which is itself not always under full control. If this prior has nothing to do with the generating process, then, for small sample size, there will be some distortion on parameter estimation or on hypothesis testing.</p>
<p>For example, I remember some analyses by Kolaczkowski and Thornton (2007, Mol Biol Evol 24:2108), suggesting that, for very short alignments, you can easily get a relatively strong support for clades that don’t exist. But this should probably be revisited. In particular, if I remember correctly, they analyzed mostly 4-taxon trees. It would be interesting to see what happens with more taxa, and under various models and priors.</p>
<p>In any case, with very small sample size, everything becomes more touchy.</p>
<p>But we are supposed to be in a “big data” era anyway, so why not instead trying to convince ourselves and other people that “single gene” analyses will always be too inconclusive to be worth the effort ?</p>
<p>Or are there still situations today where we really have to accept to work with small datasets ?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p><span class="mention">@rob_lanfear</span> I’m afraid I can’t follow your post. [quote=“rob_lanfear, post:14, topic:32”]
Let’s imagine that we have two hypotheses that generate different sets of trees of size N and M respectively.
[/quote]
OK, so N and M are real numbers.</p>
<aside class="quote no-group" data-post="14" data-topic="32" data-username="rob_lanfear">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="https://www.phylobabble.org/user_avatar/www.phylobabble.org/rob_lanfear/40/138_2.png" width="20"/> rob_lanfear:</div>
<blockquote>
<p>In a Bayesian framework we could just analyse the posterior distribution of trees and see how frequently N and M appear in the prior</p>
</blockquote>
</aside>
<p>Wait, are N and M trees or tree sets?</p>
<aside class="quote no-group" data-post="14" data-topic="32" data-username="rob_lanfear">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="https://www.phylobabble.org/user_avatar/www.phylobabble.org/rob_lanfear/40/138_2.png" width="20"/> rob_lanfear:</div>
<blockquote>
<p>Similarly, in an ML framework we could e.g. estimate the ML tree from set N and set M</p>
</blockquote>
</aside>
<p>Are they sequence data now? Sorry I can’t follow this.</p>
<p> </p>
<p>WRT the star tree paradox, may I courteously suggest that the rigoro math analysis by <a href="http://sysbio.oxfordjournals.org/content/57/4/602.short">Susko</a> and the further simulations in <a href="http://mbe.oxfordjournals.org/content/24/8/1639.long">Yang 2007</a> papers are more informative than the K&amp;T paper. Yang’s solution of employing a “data dependent prior” seems a bit strange to me, but this figure was worth a lot of words:
<img height="500" src="https://www.phylobabble.org/uploads/default/18/d256e6e6505898dc.png" width="572"/></p>
<p>That sure looks like a problem, doesn’t it seem like it to you? I don’t remember any papers saying that this goes away when such a 4 taxon tree is embedded in a larger tree (though memory is not my strong point).</p>
<p>For a solution, if it must be Bayesian, I don’t see anything other than the rjMCMC.</p>
<p> </p>
<p>We will never escape some “small data” problems: see <a href="http://phylobabble.org/t/how-to-infer-topology-and-clade-support-with-very-low-diversity-alignments/32/8?u=ematsen">my post</a> above to see an example with few informative sites from whole-genome sequencing.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/16_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">nicolas_lartill</div>
          <div class="post_content">
<p>OK, your older post is indeed convincing: there are cases where you will never have more than a few informative sites.</p>
<p>If we need to formalize and get a bit more quantitative here, for me, it would be this:</p>
<p>(1) what proportions of true/false discoveries do we get for a given number of informative sites, for a given problem and under a particular inference method (and using a given threshold) ? Ideally, one would like the methods to advertise a reasonably reliable estimate of the risk that they make such false discoveries. If they can do this even in a low-diversity regime, then users are left with sufficient information to make up their own mind.</p>
<p>(2) Then, once the rate of false discovery is at least approximately known, the second question is: for a given false discovery rate, imposed as a pre-requisite, which method gives you the most informative answer ?</p>
<p>Given that, in general, we are interested in having such good calibration properties mostly in the strong support (low false discovery rate) regime: in many practical situations, we don’t really care if nominal rates of 20% of false discoveries in fact correspond to 50%, because 20% is already too inconclusive anyway. But we do care about whether an advertised 1% really means 1% of false discovery (or at least, no more than a few %). Because these strongly supported claims will be the ones relied upon for downstream publication or decision making.</p>
<p>In any case, I think that false/true discovery rate is the crucial property we care about. And note that this is not just a small sample-size question. But it can become particularly critical under low-diversity alignment. In fact, we can further discuss about this, if you want, in a separate channel.</p>
<p>concerning polytomies, rjMCMC and the star tree paradox: I think this is a different question. Priors that deal with polytomies are important when polytomies exists in the true tree in the first place. But here, it is not that we believe that the tree is intrinsically polytomous. Instead, it is just that we do not have enough information for estimating a tree which may just be boringly bifurcating (of course, it may also be polytomous, but then it is another problem, fundamentally of model violation, not a sample size issue).</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/16_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">nicolas_lartill</div>
          <div class="post_content">
<p>Perhaps not totally explicit in my last post is what I mean by false discoveries in the present context.</p>
<p>if you take all clades with pp support &gt; 0.95, what fraction of them are in fact incorrect ? This would be your true rate of false discovery (here, given that the nominal rate is at 5%). And I would personally like this to be of the order of 5 percents (up to 10% would still be acceptable in practice, IMO).  A more stringent limit is pp&gt;0.99, for which I would expect ideally 1%, and at most a few percents, of false discoveries.</p>
<p>if this is true also for small alignments with current methods, then I would personally consider that it’s not that bad.</p>
<p>Once this is guaranteed, then the next question is: are there ways to increase sensitivity (more clades come up with support &gt;0.99) without increasing the rate of false discovery? But this should be considered as a subsidiary question.</p>
<p>In any case, SOWH, for instance, does not control for false discovery, but for type I error. And I do not think that this is really useful (see current discussions about p-values).</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Well, neat. It appears that we have wandered into unknown territory-- the upper-end calibration of low-diversity alignments. I’d love to see some simulation results.</p>
<aside class="quote" data-post="17" data-topic="32">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/nicolas_lartill/40/16_1.png" width="20"/> nicolas_lartill:</div>
<blockquote>
<p>But here, it is not that we believe that the tree is intrinsically polytomous. Instead, it is just that we do not have enough information for estimating a tree which may just be boringly bifurcating (of course, it may also be polytomous, but then it is another problem, fundamentally of model violation, not a sample size issue).</p>
</blockquote>
</aside>
<p>Imagine a scenario (modeled after the above paper on <em>Klebsiella</em> transmissions in a hospital) where perhaps because of the transfer of medical devices there were four transmissions of a microbe to a new host within minutes of each other from a clonal population. These populations then go on and evolve within the hosts and are transferred to others. How would you simulate sequences under such a scenario? I would make a polytomy for the four simultaneous transmissions.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/16_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">nicolas_lartill</div>
          <div class="post_content">
<p>oh yes sure: if good reasons to believe that there are polytomies, then one should certainly put positive prior probability on them, and rjMCMC as in Lewis Holder Holsinger 2005 is a very neat way to do this. I totally agree on that. In fact, I even tend to think that implementing LHH2005 priors on tree topologies should be more seriously considered in current programs (in particular, because we start to confront phylogenies in which there is positive evidence for radiation-like patterns).</p>
<p>I guess I was just saying that there is not reason to be <em>more</em> concerned about this issue of polytomies just because the number of informative positions is smaller: for me, these are just two independent problems.</p>
<p>yes, I think that some validation/simulation results would be nice here (and more generally)</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">GrahamJones</div>
          <div class="post_content">
<aside class="quote" data-post="1" data-topic="32">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/ematsen/40/114_1.png" width="20"/> ematsen:</div>
<blockquote>
<p>I was reading a paper recently that did a full-on BEAST analysis on what appeared to be an alignment with very few informative columns-- independent relaxed clocks, skyline plots, etc. It raised my eyebrows.</p>
</blockquote>
</aside>
<p>That sounds to me like they were using a coalescent model within a single population. In that case, the topology and node heights of the gene trees might be regarded as nuisance parameters to be integrated out, with the real interest being the population parameters. But you read the paper, I didn’t.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>They were actually interested in clade posterior probabilities.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/44_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">pscUtah</div>
          <div class="post_content">
<p>I’ve had very good luck with very short conserved intron sequences (e.g. 46 informative sites) for rather deep splits in cone snails exactly because the “fingerprint on the pipe” was so clear. The absence of conflicting sites made the phylogenetic signal super clear. So clear that my habit of using Spectral Splits was a formality.</p>
<p>I’d like to find many more of these sequences. Such seems to be the best case for very deep splits such as at the base of the eutherian tree with a comparable tree depth. Any ideas on looking not for extraordinarily long noisy sequences but for short sweet ones?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/{size}.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">aroger</div>
          <div class="post_content">
<p>Its probably not really a useful point, but the question of whether a polytomy is ‘real’ or not seems to me to depend on how you view ‘what’ the the phylogeny is representing.  I can imagine a case where there were two sequential bifurcations in a population of bacteria and the substitution rate is so low that in fact between the first and the second, there were NO fixed changes.  In this case, the phylogeny of the populations would in fact be dichotomous, but the phylogeny of the genomes is polytomous.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>