<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Weights used in opposite ways; alignments vs. phylogenetic inference</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Weights used in opposite ways; alignments vs. phylogenetic inference</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/179_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ajeffries</div>
          <div class="post_content">
<p>Apologies if this is a stupid question, I’m just trying to sanity check my understanding of how character state transformation weights are used.</p>
<p>When carrying out sequence alignments, the more common character state transformations (e.g., transitions) would be given more weight because, being more common, they are the changes you would expect to happen during evolution.</p>
<p>When carrying out a phylogenetic inference, the more common character state transformations (e.g., transitions) would be given a lower weighting because, being common, they are more likely to contain noise and are less likely to represent significant evolutionary adaptations.</p>
<p>Have I got the logic correct or wrong? I can’t say I’ve ever seen explicit documentation about how inference programs actually use their weight matrices.</p>
<p>Thanks in advance.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/117_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">argriffing</div>
          <div class="post_content">
<p>Are you referring to something like the distinction between the <a href="https://en.wikipedia.org/wiki/Models_of_DNA_evolution#K80_model_.28Kimura.2C_1980.29.5B2.5D" rel="nofollow noopener">K80</a> nucleotide transition rate parameter (usually greater than 1) vs. the <a href="http://www.clustal.org/download/clustalw_help.txt" rel="nofollow noopener">clustalw</a> nucleotide transition weight parameter (usually less than 1)?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/179_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ajeffries</div>
          <div class="post_content">
<p>Or consider the PAM tables when aligning amino acid sequences and then inferring phylogenies from them. They are empirical weight tables that show the log odds of amino acid substitutions based on observed data, It makes sense to use those numbers as positive scores when constructing an alignment. However, when using the alignment to infer a phylogeny does it make more sense to put more weight on the rare substitutions? i.e., to use the weights in an opposite manner. Or am I barking mad?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p><span class="mention">@ajeffries</span></p>
<aside class="quote" data-post="1" data-topic="589">
<div class="title">
<div class="quote-controls"></div>
<img alt="" class="avatar" height="20" src="//www.phylobabble.org/user_avatar/www.phylobabble.org/ajeffries/40/179_1.png" width="20"/> ajeffries:</div>
<blockquote>
<p>When carrying out a phylogenetic inference, the more common character state transformations (e.g., transitions) would be given a lower weighting because, being common, they are more likely to contain noise and are less likely to represent significant evolutionary adaptations.</p>
</blockquote>
</aside>
<p>Could you clarify what you mean? The common state transitions are more probable, and so have a higher value in the probability transition matrix.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/179_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ajeffries</div>
          <div class="post_content">
<p>I’ve found the answer to my question in “Molecular Systematics” 2nd ed.</p>
<p>pg. 422
“A disadvantage to the complete rejection of information on transitions [*referring to transversion parsimony] is that, while transitions may become saturated over long evolutionary distances, they may nonetheless be highly informative with respect to relationships among closely related taxa. One way around the dilemma is to assign greater weight to transversions than transitions, without going so far as to give transitions zero weight, as does transversion parsimony.”</p>
<p>pg. 424
“Because of the biochemical properties of the various amino acids, there is often little selection against changes between amino acids having similar properties (e.g., between aspartic and glutamic acids). If changes between similar residues occur very frequently, perhaps we should ignore them as well (or at least give them less weight). The generalized parsimony method can be used to implement this strategy…with the weights derived from the matrices presented by Dayhoff…or Henikoff and Henikoff.”</p>
<p>This is OPPOSITE to what would be done in an alignment process where, for instance, conservative amino acid substitutions would be favoured.</p>
<p>You’re correct, the probability transition matrix will remain the same in either use (alignments/inference) but the way the values are used are in opposite directions; high values are used as up-weights in alignment but as down-weights in inference.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>