<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Postdoc position &ldquo;Evolutionary biology of cancer&rdquo; in Cambridge UK</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Postdoc position &ldquo;Evolutionary biology of cancer&rdquo; in Cambridge UK</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/271_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">florianmarkowetz</div>
          <div class="post_content">
<p>The Markowetz lab in the Cancer Research UK Cambridge Institute at the University of Cambridge is looking for outstanding candidates to work on <strong>inferring patterns of tumor evolution</strong> from genomics data.</p>
<p>We will analyse two types of data: (1) large cohorts of single-sample bulk sequenced tumors
from the TCGA and ICGC pan-cancer projects as well as from other sources, and (2) genomes of single cells from individual tumors. We plan to adapt methods from population genetics and phylogenetics to the cancer setting. Key problems will be to compare mutation rates and
selection hotspots between the genomes of cancer clones.</p>
<p>This position is ideal for somebody trained in evolutionary biology in model systems to make the transition to biomedical applications in cancer.</p>
<p><strong>The successful applicant</strong> will have a PhD in a quantitative field like mathematics, statistics, physics, engineering, bioinformatics, or computer science. A background in evolutionary biology, molecular
evolution or population genetics is highly desired. The applicant should have a good biological background and excellent computing skills. The atmosphere at CI is very collaborative and interactive; good communication skills are key.</p>
<p><strong>To apply,</strong> please send your academic CV and a covering letter to Tania Smith at <a href="mailto:tania.smith@cruk.cam.ac.uk">tania.smith@cruk.cam.ac.uk</a></p>
<p>More about the lab at <a href="http://www.markowetzlab.org" rel="nofollow noopener">http://www.markowetzlab.org</a></p>
<p>More about tumor evolution at <a href="https://scientificbsides.wordpress.com/tag/intra-tumour-phylogeny-problem/" rel="nofollow noopener">https://scientificbsides.wordpress.com/tag/intra-tumour-phylogeny-problem/</a></p>
<p><strong>References</strong></p>
<p>Beerenwinkel et al (2014) Cancer evolution: mathematical models and computational inference, Systematic Biology.</p>
<p>Ross and Markowetz (2016), OncoNEM: Inferring tumour evolution from single-cell sequencing data, Genome Biology, 17:69</p>
<p>Schwarz et al (2015), Spatial and temporal heterogeneity in high-grade serous ovarian cancer: a phylogenetic reconstruction, PLoS Med, 12(2)</p>
<p>Yuan et al (2015), BitPhylogeny: A probabilistic framework for reconstructing intra-tumor phylogenies, Genome Biol, 16:36</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>