<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Postdoc in phylogenetics of the immune system</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Postdoc in phylogenetics of the immune system</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>I’m going to keep on advertising this position with <span class="mention">@trvrb</span> until I’m hoarse, because I think that B cell receptor analysis is just about the coolest thing right now.</p>
<p>If you need a practical reason to check this out, <em>this is a way for you to get in on the ground level of a new subfield of molecular evolution / phylogenetics which is rich in data and very weak in methods, and to get biomedical funding for phylogenetic work (which is a good thing in the current funding environment)</em>.</p>
<p>Here is the text of <a href="http://bedford.io/blog/postdoc-repertoire-dynamics/">the ad</a>:</p>
<blockquote>
<p>A new postdoctoral position is available in the groups of Trevor
Bedford and Erick Matsen at the Fred Hutchinson Cancer Research Center
located in Seattle, WA. This position will focus on analyzing immune
repertoire sequence data from an evolutionary perspective.</p>
<p>Deep sequencing of B cell repertoires has opened up the possibility of
understanding the immune response at a molecular level. This project
will analyze repertoire data to compare processes of clonal expansion
and affinity maturation in primary and secondary immune responses.
These studies will allow, for the first time, a detailed mechanistic
understanding of original antigenic sin in viral infection; results
from these studies may allow construction of vaccines that better deal
with a diversity of pathogen exposures. This work represents a
collaboration with experimental colleagues at the Fred Hutch, Emory
University and Adaptive Biotechnologies. There will be no want of
fresh data.</p>
<p>The ideal candidate will have experience with working with sequence
data and a strong interest in model-based statistical analysis of real
data sets. Data sets will comprise millions of B cell repertoire
sequences and so strong coding abilities are essential. Candidates
should have experience in at least one programming language and a
proven track-record of peer reviewed publications. Candidates with
PhDs from diverse backgrounds are encouraged to apply, including
biology, mathematics, statistics, physics and computer science.
However, experience in phylogenetics or immunological bioinformatics
is especially opportune. The Fred Hutch is an equal opportunity
employer, committed to workforce diversity. Women and minorities are
particularly encouraged to apply.</p>
<p>The position is available immediately with flexible starting dates for
a 2-year appointment with possibility of extension. Informal inquires
are welcome. Applications will be accepted until the position is
filled. The Fred Hutchinson Cancer Research Center offers competitive
salaries commensurate with experience and skills, complete with
benefits.</p>
<p>To apply please send (1) cover letter that includes the names and
contacts for three references and a short statement of research
interests, (2) a current CV and (3) code samples or links to
published/distributed code to <a href="mailto:trevor@bedford.io">trevor@bedford.io</a>.</p>
</blockquote>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>In related news, we also have a postdoc open, <a href="http://matsen.fredhutch.org/general/2015/09/01/ab-postdoc.html" rel="nofollow noopener">posted Tuesday on our website</a>:</p>
<h3> Postdoctoral position to study molecular evolution and phylogenetics of immune cells </h3>
<p><i> 01 Sep 2015, by Erick </i></p>
<div>
<p><img height="450" src="https://www.phylobabble.org/uploads/default/original/1X/9de477118a51fff779b65315de5cc57a5d82d605.png" width="514"/>
</p>
<p>Our adaptive immune systems continually update themselves to neutralize and destroy pathogens.
The receptor sequences of antibody-making B cells undergo a Darwinian process of mutation and selection which improves their binding to antigen.<sup><a href="#fn:1">1</a></sup>
It is now possible to sequence these B cell receptors (BCRs) in high throughput, giving a profound new perspective on how the immune system responds to infection.<sup><a href="#fn:2">2</a></sup>
Although the elements of B cell affinity maturation are the same as molecular evolution in other settings, being based on recombination, point mutation, and selection, there are a many important differences.
These differences, along with the volume of sequence data available, bring new challenges for phylogenetics and molecular evolution.</p>
<p>The translational medical consequences of improved methods will be significant.
Improved methods will especially help in understanding the development of <a href="http://dx.doi.org/10.1016/j.cell.2015.03.004" rel="nofollow noopener">broadly neutralizing antibodies against HIV</a>.
The current best hope for an effective HIV vaccine is to try to elicit such antibodies, but in order to do so we need to understand in detail how such antibodies.
However, the effective antibodies naturally produced in adults require many years of mutation from an unlikely starting point.
Thus they will be exceedingly difficult to elicit with a vaccine.</p>
<p>Fortunately, <a href="http://research.fredhutch.org/overbaugh/en.html" rel="nofollow noopener">Julie Overbaugh</a>’s group here at the Fred Hutch recently published <a href="http://www.nature.com/nm/journal/v20/n6/full/nm.3565.html" rel="nofollow noopener">a landmark paper</a>, in which they found broadly neutralizing antibodies in HIV-positive infants.
These antibodies mature in significantly less time than in adults (above graphic from a <a href="http://www.nature.com/nm/journal/v20/n6/full/nm.3598.html" rel="nofollow noopener">News &amp; Views about the paper</a>).
The maturation pathway of these antibodies is not yet understood.</p>
<p>This open postdoc position is to develop new methods for B cell receptor sequence analysis and apply them to sequences from the Overbaugh lab infant study.
Our group is dedicated to bringing modern model-based statistical<sup><a href="#fn:3">3</a></sup> molecular evolution and phylogenetics approaches to BCR sequence analysis that scale to large data sets.
Thus far we have developed means of doing <a href="http://dx.doi.org/10.1098/rstb.2014.0244" rel="nofollow noopener">selection inference</a> on B cell receptors, <a href="http://arxiv.org/abs/1503.04224" rel="nofollow noopener">VDJ annotation</a> (in press, PLOS Comp Bio), and clonal family inference (paper out soon; see <a href="http://matsen.github.io/talks/bcell.html#/" rel="nofollow noopener">my talk</a>).
Having made progress in these arenas, we are now starting to improve phylogenetic methods for B cell receptors.</p>
<p>We would like someone who is strong in computational statistics and/or molecular evolution, but who is motivated by and engaged with the underlying biology.
This project will involve significant computer programming and data analysis, but no wet lab work.
The methods-development part of this postdoc will be joint with <a href="http://bedford.io" rel="nofollow noopener">Trevor Bedford</a>, <a href="http://www.stat.washington.edu/vminin/" rel="nofollow noopener">Vladimir Minin</a>, and Harlan Robins, co-founder of <a href="http://www.adaptivebiotech.com" rel="nofollow noopener">Adaptive Biotechnologies</a>.</p>
<p>The position will come with a competitive postdoc-level salary with great benefits for two years.
There is some possibility of extension.
Fred Hutchinson Cancer Research Center, home of about 190 faculty including three Nobel laureates, is an independent, nonprofit research institution dedicated to the development and advancement of biomedical research.
The environment is lively yet casual, with a strong emphasis on collaborative work.
The Center is housed in a lovely campus next to Lake Union a short walk from downtown, and a slightly longer walk from the University of Washington.
Powerful computing resources and a helpful IT staff await.</p>
<p>If you are interested in this position, please send Erick some representative publications, code samples, and a CV.</p>
<p><em>If you are interested more in evolutionary dynamics of B cells please see <a href="http://bedford.io/blog/postdoc-repertoire-dynamics/" rel="nofollow noopener">the related opening on Trevor’s website</a>.</em></p>
<hr/>
<div>
<ol>
<li>
<p><a href="http://dx.doi.org/10.1098/rstb.2014.0235" rel="nofollow noopener">Cobey et al, 2015</a> <a href="#fnref:1">↩</a></p>
</li>
<li>
<p><a href="http://dx.doi.org/10.1038/nbt.2782" rel="nofollow noopener">Georgiou et al, 2014</a> <a href="#fnref:2">↩</a></p>
</li>
<li>
<p>Here <em>model-based</em> means that our ideas about how the system works are formalized into a probabilistic model, and <em>statistical inference</em> means that uncertainty at every stage is pushed through to obtain parameter estimates with appropriate statements of confidence in those estimates. <a href="#fnref:3">↩</a></p>
</li>
</ol>
</div>
</div>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>