<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>Paper: An experimental phylogeny to benchmark ancestral sequence reconstruction</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">Paper: An experimental phylogeny to benchmark ancestral sequence reconstruction</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>I thought this was a fun read that might have escaped readers of the usual journals:</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/27628687" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/27628687" rel="nofollow noopener" target="_blank">An experimental phylogeny to benchmark ancestral sequence reconstruction.</a>
</h4>
<div class="date">
  RN Randall, CE Radford, KA Roof, DK Natarajan and EA Gaucher,
  <i>Nature communications</i>, Sep 2016 15
</div>
<p class="pubmed-abstract">
  Ancestral sequence reconstruction (ASR) is a still-burgeoning method that has revealed many key mechanisms of molecular evolution. One criticism of the approach is an inability to validate its algorithms within a biological context as opposed to a computer simulation. Here we build an experimental phylogeny using the gene of a single red fluorescent protein to address this criticism. The evolved phylogeny consists of 19 operational taxonomic units (leaves) and 17 ancestral bifurcations (nodes) that display a wide variety of fluorescent phenotypes. The 19 leaves then serve as 'modern' sequences that we subject to ASR analyses using various algorithms and to benchmark against the known ancestral genotypes and ancestral phenotypes. We confirm computer simulations that show all algorithms infer ancient sequences with high accuracy, yet we also reveal wide variation in the phenotypes encoded by incorrectly inferred sequences. Specifically, Bayesian methods incorporating rate variation significantly outperform the maximum parsimony criterion in phenotypic accuracy. Subsampling of extant sequences had minor effect on the inference of ancestral sequences.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>H/T my visiting student Kristian Davidsen.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>