<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>DNA or protein sequence identity vs true phylogenetic distance</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">DNA or protein sequence identity vs true phylogenetic distance</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/375_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">BrianFoley</div>
          <div class="post_content">
<p><img height="500" src="https://www.phylobabble.org/uploads/default/60/640066b934ed987e.jpg" width="318"/> </p>
<p>One quick and easy tool for visualizing the DNA distances in a data set, is the DAMBE function under graphics to plot the transitions and transversions vs F84 phylogenetic distance for each pairwise comparison in the data set.  I have attached a plot here, showing the plots for Primate T-Cell Leukemia Viruses and Primate Lentiviruses.  We know that HIV-1 M group, with F84 distances less than 0.15 here, represent roughly 100 years of evolution.  But comparing HIV-1 to SIV from African Green Monkey (distances &gt; 0.5) the age estimate to the common ancestor is in the <a href="http://www.plospathogens.org/article/info:doi/10.1371/journal.ppat.1003135#ppat-1003135-g004" rel="noopener nofollow ugc">millions of years range</a>.</p>
<p>Using DNA or protein distances, we don’t have any methods as far as I know, that would extrapolate from 100 years to get 15% (phylogenetically corrected) distance to more than 100,000 years for 50% (phylogenetically corrected) distance, let alone millions of years.  Silent sites become more than saturated with mutations while at the same time many other sites remain absolutely invariant over time.  Thus, calibrating the “molecular clock” has to be done in a time/distance range that is applicable to the data.</p>
<p>The same plots can be done for mammals, vertebrates, etc.   The DNA distances between the most distant mammals is more than saturated with mutation in mitochondrial DNA, but not nuclear genes.  DNA distances in nuclear genes become saturated when comparing vertebrates (fish, amphibians, reptiles, birds, mammals etc) and at those distances the mitochondrial genomes are easy to align but quite misleading for the “molecular clock” methods.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>