<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>[Paper] The Genealogical Population Dynamics of HIV-1 in a Large Transmission Chain: Bridging within and among Host Evolutionary Rates</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">[Paper] The Genealogical Population Dynamics of HIV-1 in a Large Transmission Chain: Bridging within and among Host Evolutionary Rates</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Very interesting work from <span class="mention">@arambaut</span>, <span class="mention">@alexei_drummond</span>, <span class="mention">@guy_baele</span>, and <span class="mention">@philippe_Lemey</span>.</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24699231" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24699231" rel="nofollow noopener" target="_blank">The genealogical population dynamics of HIV-1 in a large transmission chain: bridging within and among host evolutionary rates.</a>
</h4>
<div class="date">
  B Vrancken, A Rambaut, MA Suchard, A Drummond, G Baele, I Derdelinckx, E Van Wijngaerden, AM Vandamme, K Van Laethem and P Lemey,
  <i>PLoS computational biology</i>, Apr 2014
</div>
<p class="pubmed-abstract">
  Transmission lies at the interface of human immunodeficiency virus type 1 (HIV-1) evolution within and among hosts and separates distinct selective pressures that impose differences in both the mode of diversification and the tempo of evolution. In the absence of comprehensive direct comparative analyses of the evolutionary processes at different biological scales, our understanding of how fast within-host HIV-1 evolutionary rates translate to lower rates at the between host level remains incomplete. Here, we address this by analyzing pol and env data from a large HIV-1 subtype C transmission chain for which both the timing and the direction is known for most transmission events. To this purpose, we develop a new transmission model in a Bayesian genealogical inference framework and demonstrate how to constrain the viral evolutionary history to be compatible with the transmission history while simultaneously inferring the within-host evolutionary and population dynamics. We show that accommodating a transmission bottleneck affords the best fit our data, but the sparse within-host HIV-1 sampling prevents accurate quantification of the concomitant loss in genetic diversity. We draw inference under the transmission model to estimate HIV-1 evolutionary rates among epidemiologically-related patients and demonstrate that they lie in between fast intra-host rates and lower rates among epidemiologically unrelated individuals infected with HIV subtype C. Using a new molecular clock approach, we quantify and find support for a lower evolutionary rate along branches that accommodate a transmission event or branches that represent the entire backbone of transmitted lineages in our transmission history. Finally, we recover the rate differences at the different biological scales for both synonymous and non-synonymous substitution rates, which is only compatible with the 'store and retrieve' hypothesis positing that viruses stored early in latently infected cells preferentially transmit or establish new infections upon reactivation.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>Methodologically, this paper provides long-overdue inferential methods that distinguish (the rather different) within-host and between-host evolutionary processes, as well as integrate epidemiological information. I was wondering if the authors could say if this method would improve inference with data sets for which we have no epidemiological information, just stratification of sequences by host.</p>
<p>Biologically, it has been known for a while that viruses from chronic infections have certain mutations that are not present in primary infection. Here the authors provide further evidence that infectious HIV derives from “stored” virions rather than mutated virion lineages which have reverted. This builds on the work of:</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23639104" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23639104" rel="nofollow noopener" target="_blank">Within-host and between-host evolutionary rates across the HIV-1 genome.</a>
</h4>
<div class="date">
  S Alizon and C Fraser,
  <i>Retrovirology</i>, May 2013 02
</div>
<p class="pubmed-abstract">
  HIV evolves rapidly at the epidemiological level but also at the within-host level. The virus' within-host evolutionary rates have been argued to be much higher than its between-host evolutionary rates. However, this conclusion relies on analyses of a short portion of the virus envelope gene. Here, we study in detail these evolutionary rates across the HIV genome.We build phylogenies using a relaxed molecular clock assumption to estimate evolutionary rates in different regions of the HIV genome. We find that these rates vary strongly across the genome, with higher rates in the envelope gene (env). Within-host evolutionary rates are consistently higher than between-host rates throughout the HIV genome. This difference is significantly more pronounced in env. Finally, we find weak differences between overlapping and non-overlapping regions.We provide a genome-wide overview of the differences in the HIV rates of molecular evolution at the within- and between-host levels. Contrary to hepatitis C virus, where differences are only located in the envelope gene, within-host evolutionary rates are higher than between-host evolutionary rates across the whole HIV genome. This supports the hypothesis that HIV strains that are less adapted to the host have an advantage during transmission. The most likely mechanism for this is storage and then preferential transmission of viruses in latent T-cells. These results shed a new light on the role of the transmission bottleneck in the evolutionary dynamics of HIV.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>

          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Here is a note from the first author Bram Vrancken from <span class="mention">@philippe_Lemey</span>’s group responding to my question:</p>
<blockquote>
<p>We think modeling the transmission bottleneck does not really improve inferences, and certainly not when it cannot be pinpointed who infected who (a situation in which we cannot specify the transmission model and thus fall back on other standard coalescent models). If there is improvement, most of it will likely come from the rate modeling (without epidemiological info we can set up a similar backbone effect model between patients): the improved rate variation modeling should in principle result in more accurate inferences on the divergence times and perhaps even clustering. I’ve discussed this with Philippe and he doesn’t expect a large effect for both though.</p>
</blockquote>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>I just wanted to note that the above paper has interesting consequences for this one:</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24874208" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24874208" rel="nofollow noopener" target="_blank">Timing and order of transmission events is not directly reflected in a pathogen phylogeny.</a>
</h4>
<div class="date">
  E Romero-Severson, H Skar, I Bulla, J Albert and T Leitner,
  <i>Molecular biology and evolution</i>, Sep 2014
</div>
<p class="pubmed-abstract">
  Pathogen phylogenies are often used to infer spread among hosts. There is, however, not an exact match between the pathogen phylogeny and the host transmission history. Here, we examine in detail the limitations of this relationship. First, all splits in a pathogen phylogeny of more than 1 host occur within hosts, not at the moment of transmission, predating the transmission events as described by the pretransmission interval. Second, the order in which nodes in a phylogeny occur may be reflective of the within-host dynamics rather than epidemiologic relationships. To investigate these phenomena, motivated by within-host diversity patterns, we developed a two-phase coalescent model that includes a transmission bottleneck followed by linear outgrowth to a maximum population size followed by either stabilization or decline of the population. The model predicts that the pretransmission interval shrinks compared with predictions based on constant population size or a simple transmission bottleneck. Because lineages coalesce faster in a small population, the probability of a pathogen phylogeny to resemble the transmission history depends on when after infection a donor transmits to a new host. We also show that the probability of inferring the incorrect order of multiple transmissions from the same host is high. Finally, we compare time of HIV-1 infection informed by genetic distances in phylogenies to independent biomarker data, and show that, indeed, the pretransmission interval biases phylogeny-based estimates of when transmissions occurred. We describe situations where caution is needed not to misinterpret which parts of a phylogeny that may indicate outbreaks and tight transmission clusters.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>as in changes our interpretation of what coalescent trees look like in transmission.</p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>