<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width">
    <title>[Paper] Hypermutable DNA chronicles the evolution of human colon cancer</title>
    <link rel="stylesheet" href="../../../archived.css" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\\\(','\\\\)']]}});
    </script>
    <script type="text/javascript" async
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
  </head>

  <body>
    <header class="header">
      <div class="title-span">
        <a href="../../../">
          <img src="../../../images/site-logo.png" height="40" alt="phylobabble" id="site-logo" />
        </a>
      </div>
    </header>

    <div class="main">
    <div class="archive-span">Archive 04/01/2022.</div>
    <h1 class="topic-title">[Paper] Hypermutable DNA chronicles the evolution of human colon cancer</h1>
            <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<aside class="onebox allowlistedgeneric" data-onebox-src="https://pubmed.ncbi.nlm.nih.gov/24753616/">
<header class="source">
<img class="site-icon" height="64" src="https://cdn.ncbi.nlm.nih.gov/coreutils/nwds/img/favicons/favicon.ico" width="64"/>
<a href="https://pubmed.ncbi.nlm.nih.gov/24753616/" rel="noopener" target="_blank">PubMed</a>
</header>
<article class="onebox-body">
<img class="thumbnail onebox-avatar" data-small-upload="https://www.phylobabble.org/uploads/default/optimized/1X/663a2ec8b6b28f6362f65026ac415550ee304255_2_10x10.png" height="500" src="https://www.phylobabble.org/uploads/default/optimized/1X/663a2ec8b6b28f6362f65026ac415550ee304255_2_500x500.png" srcset="https://www.phylobabble.org/uploads/default/optimized/1X/663a2ec8b6b28f6362f65026ac415550ee304255_2_500x500.png, https://www.phylobabble.org/uploads/default/optimized/1X/663a2ec8b6b28f6362f65026ac415550ee304255_2_750x750.png 1.5x, https://www.phylobabble.org/uploads/default/optimized/1X/663a2ec8b6b28f6362f65026ac415550ee304255_2_1000x1000.png 2x" width="500"/>
<h3><a href="https://pubmed.ncbi.nlm.nih.gov/24753616/" rel="noopener" target="_blank">Hypermutable DNA chronicles the evolution of human colon cancer - PubMed</a></h3>
<p>Intratumor genetic heterogeneity reflects the evolutionary history of a cancer and is thought to influence treatment outcomes. Here we report that a simple PCR-based assay interrogating somatic variation in hypermutable polyguanine (poly-G) repeats...</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>Here’s an example figure, but note that the “alignment” is transpose of how we usually think.</p>
<p><img height="426" src="https://www.phylobabble.org/uploads/default/78/b37c86ccb87966da.png" width="452"/> </p>
<pre><code>Phylogenetic Reconstruction. We reconstructed phylogenies using two in-
dependent approaches. First, we calculated a distance matrix for each patient
using an “equal or not” distance (31). This method increases the distances
between two samples if they have unequal genotypes, regardless of the
magnitude of the difference. We then used neighbor-joining (51) in R to
infer the phylogenetic relationships between samples. In the very rare case
of missing values, we imputed them using the nearest neighbor. We used
bootstrapping with 1,000 replicates to test the reliability of the resulting
trees (52) and collapsed all interior branches with bootstrap values below
70% into polytomies. Next, we used Bayesian inference of phylogeny—
a methodology that relies on a fundamentally different set of principles
than neighbor-joining—to construct the phylogenies. The results were al-
most identical in all cases, confirming the robustness of our approach.
Bayesian phylogenies and posterior probability values for all clades are
presented in SI Appendix, Fig. S10. We used the software MrBayes (53) with
the same model parameters that were previously used for the analysis of
poly-G tract mutation profiles (21).</code></pre>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>And another paper about phylogenetics and cell lineages, but with a different data type.</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24743184" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24743184" rel="nofollow noopener" target="_blank">Phylogenetic quantification of intra-tumour heterogeneity.</a>
</h4>
<div class="date">
  RF Schwarz, A Trinh, B Sipos, JD Brenton, N Goldman and F Markowetz,
  <i>PLoS computational biology</i>, Apr 2014
</div>
<p class="pubmed-abstract">
  Intra-tumour genetic heterogeneity is the result of ongoing evolutionary change within each cancer. The expansion of genetically distinct sub-clonal populations may explain the emergence of drug resistance, and if so, would have prognostic and predictive utility. However, methods for objectively quantifying tumour heterogeneity have been missing and are particularly difficult to establish in cancers where predominant copy number variation prevents accurate phylogenetic reconstruction owing to horizontal dependencies caused by long and cascading genomic rearrangements. To address these challenges, we present MEDICC, a method for phylogenetic reconstruction and heterogeneity quantification based on a Minimum Event Distance for Intra-tumour Copy-number Comparisons. Using a transducer-based pairwise comparison function, we determine optimal phasing of major and minor alleles, as well as evolutionary distances between samples, and are able to reconstruct ancestral genomes. Rigorous simulations and an extensive clinical study show the power of our method, which outperforms state-of-the-art competitors in reconstruction accuracy, and additionally allows unbiased numerical quantification of tumour heterogeneity. Accurate quantification and evolutionary inference are essential to understand the functional consequences of tumour heterogeneity. The MEDICC algorithms are independent of the experimental techniques used and are applicable to both next-generation sequencing and array CGH data.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>I think it would be fun to have a phyloseminar on these papers. Opinions?</p>
<p><span class="mention">@mathmomike</span> — has there been any recent work from the “tree of cells” people?</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>If anyone else is interested in following this topic, there has been an interesting recent contribution:</p>
<aside class="onebox pubmed">
<header class="source">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25010360" rel="nofollow noopener" target="_blank">ncbi.nlm.nih.gov</a>
</header>
<article class="onebox-body">
<h4>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25010360" rel="nofollow noopener" target="_blank">Inferring clonal composition from multiple sections of a breast cancer.</a>
</h4>
<div class="date">
  H Zare, J Wang, A Hu, K Weber, J Smith, D Nickerson, C Song, D Witten, CA Blau and WS Noble,
  <i>PLoS computational biology</i>, Jul 2014
</div>
<p class="pubmed-abstract">
  Cancers arise from successive rounds of mutation and selection, generating clonal populations that vary in size, mutational content and drug responsiveness. Ascertaining the clonal composition of a tumor is therefore important both for prognosis and therapy. Mutation counts and frequencies resulting from next-generation sequencing (NGS) potentially reflect a tumor's clonal composition; however, deconvolving NGS data to infer a tumor's clonal structure presents a major challenge. We propose a generative model for NGS data derived from multiple subsections of a single tumor, and we describe an expectation-maximization procedure for estimating the clonal genotypes and relative frequencies using this model. We demonstrate, via simulation, the validity of the approach, and then use our algorithm to assess the clonal composition of a primary breast cancer and associated metastatic lymph node. After dividing the tumor into subsections, we perform exome sequencing for each subsection to assess mutational content, followed by deep sequencing to precisely count normal and variant alleles within each subsection. By quantifying the frequencies of 17 somatic variants, we demonstrate that our algorithm predicts clonal relationships that are both phylogenetically and spatially plausible. Applying this method to larger numbers of tumors should cast light on the clonal evolution of cancers in space and time.
</p>
</article>
<div class="onebox-metadata">
</div>
<div style="clear: both"></div>
</aside>
<p>which really takes a completely different approach. Rather than focusing on phylogenetic inference itself, this new work focuses on inferring the <em>posterior</em> on per-site clonal genotypes in a mixed sample. They then build phylogenetic trees by either manually solving the perfect phylogeny problem or, if that is not possible, finding “nearby” genotype inferences for which the perfect phylogeny problem is soluble.</p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>Related papers keep coming!</p>
<p><a class="onebox" href="http://genome.cshlp.org/content/early/2014/07/24/gr.180281.114.abstract" target="_blank">http://genome.cshlp.org/content/early/2014/07/24/gr.180281.114.abstract</a></p>
<blockquote>
<p>TITAN: Inference of copy number architectures in clonal cell
populations from tumor whole genome sequence data Gavin Ha1, Andrew
Roth1, Jaswinder Khattra1, Julie Ho2, Damian Yap1, Leah M Prentice2,
Nataliya Melnyk2, Andrew McPherson1, Ali Bashashati1, Emma Laks1,
Justina Biele1, Jiarui Ding1, Alan Le1, Jamie Rosner1, Karey
Shumansky1, Marco A Marra3, C Blake Gilks4, David G Huntsman2, Jessica
N McAlpine5, Samuel Aparicio1 and Sohrab P Shah1,6</p>
<ul>
<li>Author Affiliations</li>
</ul>
<p>1 BC Cancer Agency; 2 Centre for Translational and Applied Genomics; 3
Genome Sciences Centre; 4 Vancouver General Hospital; 5 University of
British Columbia ↵* Corresponding author; email: sshah@bccrc.ca
Abstract</p>
<p>The evolution of cancer genomes within a single tumor creates mixed
cell populations with divergent somatic mutational landscapes.
Inference of tumor subpopulations has been disproportionately focused
on the assessment of somatic point mutations, whereas computational
methods targeting evolutionary dynamics of copy number alterations
(CNA) and loss of heterozygosity (LOH) in whole genome sequencing data
remain under-developed. We present a novel probabilistic model, TITAN,
to infer CNA and LOH events while accounting for mixtures of cell
populations, thereby estimating the proportion of cells harboring each
event. We evaluate TITAN on idealized mixtures, simulating clonal
populations from whole genome sequences taken from genomically
heterogeneous ovarian tumor sites collected from the same patient. In
addition, we show in 23 whole genomes of breast tumors that inference
of CNA and LOH using TITAN critically inform population structure and
the nature of the evolving cancer genome. Finally, we experimentally
validated subclonal predictions using fluorescence in situ
hybridization (FISH) and single-cell sequencing from an ovarian cancer
patient sample, thereby recapitulating the key modeling assumptions of
TITAN.</p>
</blockquote>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>And another!</p>
<blockquote>
<p>BayClone: Bayesian Nonparametric Inference of Tumor Subclones Using
NGS Data Subhajit Sengupta1 , Jin Wang2 , Juhee Lee3 , Peter Muller4
, Kamalakar Gulukota5 , Arunava Banerjee6 , Yuan Ji1;7; 1Center for
Biomedical Research Informatics, NorthShore University HealthSystem
2Department of Statistics, University of Illinois at Urbana-Champaign
3Department of Applied Mathematics and Statistics, University of
California Santa Cruz 4Department of Mathematics, University of Texas
Austin 5Center for Molecular Medicine, NorthShore University
HealthSystem 6Department of Computer &amp; Information Science &amp;
Engineering, University Of Florida 7Department of Health Studies, The
University Of Chicago In this paper, we present a novel feature
allocation model to describe tumor heterogeneity (TH) using
next-generation sequencing (NGS) data. Taking a Bayesian approach, we
extend the Indian buet process (IBP) to dene a class of
nonparametric models, the categorical IBP (cIBP). A cIBP takes
categorical values to denote homozygous or heterozygous genotypes at
each SNV. We dene a subclone as a vector of these categorical values,
each corresponding to an SNV. Instead of partitioning somatic
mutations into non-overlapping clusters with similar cellular
prevalences, we took a dierent approach using feature allocation.
Importantly, we do not assume somatic mutations with similar cellular
prevalence must be from the same subclone and allow overlapping
mutations shared across subclones. We argue that this is closer to the
underlying theory of phylogenetic clonal expansion, as somatic
mutations occurred in parent subclones should be shared across the
parent and child subclones. Bayesian inference yields posterior
probabilities of the number, genotypes, and proportions of subclones
in a tumor sample, thereby providing point estimates as well as
variabilities of the estimates for each subclone. We report results on
both simulated and real data. BayClone is available at
<a href="http://health.bsd.uchicago.edu/yji/soft.html">http://health.bsd.uchicago.edu/yji/soft.html</a>.</p>
</blockquote>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>And another!</p>
<blockquote>
<p>Comparing Nonparametric Bayesian Tree Priors for Clonal Reconstruction
of Tumors</p>
<p>Amit G. Deshwar, Shankar Vembu, Quaid Morris</p>
<p>Statistical machine learning methods, especially nonparametric
Bayesian methods, have become increasingly popular to infer clonal
population structure of tumors. Here we describe the treeCRP, an
extension of the Chinese restaurant process (CRP), a popular
construction used in nonparametric mixture models, to infer the
phylogeny and genotype of major subclonal lineages represented in the
population of cancer cells. We also propose new split-merge updates
tailored to the subclonal reconstruction problem that improve the
mixing time of Markov chains. In comparisons with the tree-structured
stick breaking prior used in PhyloSub, we demonstrate superior mixing
and running time using the treeCRP with our new split-merge
procedures. We also show that given the same number of samples, TSSB
and treeCRP have similar ability to recover the subclonal structure of
a tumor.</p>
</blockquote>
<p><a class="onebox" href="http://arxiv.org/abs/1408.2552" target="_blank">http://arxiv.org/abs/1408.2552</a></p>
          </div>
        </div>
      </div>

      <div class="post_container">
        <div class="avatar_container">
          <img src="../../../images/114_2.png" class="avatar" />
        </div>
        <div class="post">
          <div class="user_name">ematsen</div>
          <div class="post_content">
<p>… !!!</p>
<blockquote>
<p>Bayesian Inference for Tumor Subclones Accounting for Sequencing and
Structural Variants</p>
<p>Juhee Lee, Peter Mueller, Subhajit Sengupta, Kamalakar Gulukota, Yuan
Ji</p>
<p>Tumor samples are heterogeneous. They consist of different subclones
that are characterized by differences in DNA nucleotide sequences and
copy numbers on multiple loci. Heterogeneity can be measured through
the identification of the subclonal copy number and sequence at a
selected set of loci. Understanding that the accurate identification
of variant allele fractions greatly depends on a precise determination
of copy numbers, we develop a Bayesian feature allocation model for
jointly calling subclonal copy numbers and the corresponding allele
sequences for the same loci. The proposed method utilizes three random
matrices, L, Z and w to represent subclonal copy numbers (L), numbers
of subclonal variant alleles (Z) and cellular fractions of subclones
in samples (w), respectively. The unknown number of subclones implies
a random number of columns for these matrices. We use next-generation
sequencing data to estimate the subclonal structures through inference
on these three matrices. Using simulation studies and a real data
analysis, we demonstrate how posterior inference on the subclonal
structure is enhanced with the joint modeling of both structure and
sequencing variants on subclonal genomes. Software is available at
this http URL</p>
</blockquote>
<p><a class="onebox" href="http://arxiv.org/abs/1409.7158" target="_blank">http://arxiv.org/abs/1409.7158</a></p>
          </div>
        </div>
      </div>


    </div>
  </body>
</html>